sig/ffi/function.rbs

```diff
@@ -1,6 +1,7 @@
 module FFI
   interface _Function
-    def attach: (Module mod, String name) -> self
+    # TODO: leads to a endless recursion when used with -rrbs/test/setup
+    # def attach: (Module mod, String name) -> self
```

Comment:
Perhaps porting `Function#attach` to Ruby (similar to its cousin, `VariadicInvoker#attach`) and flatten the `RegisterAttach` private module away would solve this problem at its root.

> Did you know that even CRuby is migrating some implementations from C to Ruby?
> With all the JIT improvements, Ruby code’s performance is rivaling that of C itself!

In fact, it makes sense to me if `Function` and `VariadicInvoker` share some implementations with a common ancestor module, or even (if past design decisions aren’t in the way) have `VariadicInvoker` inherit from `Function`.

Reply:
I opened an issue regarding the [stack level too deep](https://github.com/ruby/rbs/issues/1710).